RIOT-OS/RIOT
# cpu/stm32:  get the cpuid address from the UID_BASE constant defined in CMSIS

### Contribution description

This PR is an attempt to remove the hardcoded cpuid address defined for each STM32 families (and sometimes for a specific model).
Some CMSIS headers were a bit outdated so had to be updated: all f2xx, f103xx, f334, f72x.

While updating the CMSIS, I also noticed that all CCR registers for timer peripherals have to be patched. I'd like to revert this and adapt the timer peripheral driver accordingly.




### Testing procedure

- A green Murdock
- Verify that `tests/periph_cpuid` is still working and return the same result compared to master
- Maybe also run `compile_and_test_for_board.py` on some of the affected STM32 boards.




### Issues/PRs references

None
